distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/platform/TestConfiguration.java

```diff
@@ -533,231 +333,568 @@ public void testExportAfterDeletingAConfiguration() throws Exception {
+    }
+
+    private Path getExportDirectory() {
+        return Paths.get(ddfHome, "etc", "exported");
+    }
+
+    private Path getExportSubDirectory(String... paths) {
+        Path directory = getExportDirectory().resolve("etc");
+
+        for (String path : paths) {
+            directory = directory.resolve(path);
+        }
+
+        return directory;
+    }
+
+    private void addConfigurationFile(String fileName, InputStream inputStream) throws IOException {
+        FileUtils.copyInputStreamToFile(inputStream, getPathToConfigDirectory().resolve(fileName)
+                .toFile());
+        inputStream.close();
+    }
+
+    private void addConfigurationFileAndWaitForSuccessfulProcessing(String resourceName,
+            InputStream inputStream) throws IOException {
+        addConfigurationFile(resourceName, inputStream);
+
+        expect("File " + getPathToProcessedDirectory().resolve(resourceName).toString() + " exists")
+                .within(20, SECONDS).until(
+                        () -> Files.exists(getPathToProcessedDirectory().resolve(resourceName)));
+    }
+
+    private void addConfigurationFileAndWaitForFailedProcessing(String resourceName,
+            InputStream inputStream) throws IOException {
+        addConfigurationFile(resourceName, inputStream);
+
+        expect("File " + getPathToFailedDirectory().resolve(resourceName).toString() + " exists").within(20, SECONDS).until(
+                () -> Files.exists(getPathToFailedDirectory().resolve(resourceName)));
+    }
+
+    private InputStream replaceTextInResource(InputStream is, String textToReplace,
+            String replacement) throws IOException {
+        StringWriter writer = new StringWriter();
+        IOUtils.copy(is, writer);
+        String original = writer.toString();
+        String modified = original.replace(textToReplace, replacement);
+        return IOUtils.toInputStream(modified, "UTF-8");
+    }
+    
+    private void assertThatDirectoryContains(Path path, String... fileNames) {
+        String[] keystoreFiles = path.toFile()
+                .list();
+        assertThat("Exported files should not be null.", keystoreFiles, is(notNullValue()));
+        assertThat(String.format("Files missing in %s directory", path.toString()),
+                keystoreFiles,
+                arrayContainingInAnyOrder(fileNames));
+    }
+    
+    /**
+     * Class that provides utility and assertion methods for a Managed Service Felix configuration
+     * file.
+     * <p/>
+     * Note: Since we have custom code in the @{link FelixPersistenceStrategy} class to convert
+     * floats and doubles, we cannot simply rely on what we read from the file to assert that the
+     * Configuration object has the right properties. For this reason, we need to create our own
+     * Dictionary of properties in {@link #getExpectedProperties()} and use that when doing our
+     * assertions.
+     */
+    private static class ManagedServiceConfigFile {
```

Comment:
I thought about that, but I like them in the TestConfiguration file since they are really private to the test class.  I did move them to the bottom of the class so they are out of the way.